### PR TITLE
Routes:  Move into Frame2.tsx

### PR DESCRIPTION
**Pull request by GordonSmith**
Add custom History manager
Add better URL "hacking" to Workunits

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by GordonSmith** [comment]
@miguelvazq @kunalaswani please review

File: esp/src/src/react/components/Frame2.tsx [hunk 1]
```diff
@@ -98,32 +98,38 @@ const useStyles = makeStyles((theme) => ({
...
-export const Frame2: React.FunctionComponent<Frame2> = ({
-    widgetClassID = "ActivityWidget",
-    widgetParams
-}) => {
+export const Frame2: React.FunctionComponent<Frame2> = () => {
     const classes = useStyles();
+
     const mainRef = React.useRef<HTMLDivElement>();
 
+    const [body, setBody] = React.useState(<h1>...loading...</h1>);
```
Comment (miguelvazq) [inline esp/src/src/react/components/Frame2.tsx:108]: I added the circle progress previously but it was removed? https://github.com/GordonSmith/HPCC-Platform/blob/98347ee87a3109654ad824eb6fc52bff9ed8a1fd/esp/src/src/react/components/Frame.tsx#L69

File: esp/src/src/react/components/routes.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import * as React from "react";
```
[the remark this answers is not in the document]
Comment (GordonSmith) [reply esp/src/src/react/components/routes.tsx:1]: Yes - all the actions return JSX structures.

File: esp/src/src/react/components/Frame2.tsx (hunk 1 above)
Comment (GordonSmith) [reply esp/src/src/react/components/Frame2.tsx:108]: Yes - was trying to get back to basics for the sizing.  Adding it back should be easy enough.

**Pull request merged**